packages/mui-material/src/Select/Select.d.ts

```diff
@@ -152,6 +128,38 @@ export interface SelectProps<T = unknown>
   variant?: 'standard' | 'outlined' | 'filled';
 }
 
+type ConditionalRenderValueType<T> =
+  | {
+      /**
+       * If `true`, a value is displayed even if no items are selected.
+       *
+       * In order to display a meaningful value, a function can be passed to the `renderValue` prop which
+       * returns the value to be displayed when no items are selected.
+       *
+       * ⚠️ When using this prop, make sure the label doesn't overlap with the empty displayed value.
+       * The label should either be hidden or forced to a shrunk state.
+       * @default false
+       */
+      displayEmpty?: false;
+      /**
+       * If `true`, `value` must be an array and the menu will support multiple selections.
+       * @default false
+       */
+      multiple?: boolean;
+      /**
+       * Render the selected value.
+       * You can only use it when the `native` prop is `false` (default).
+       *
+       * @param {any} value The `value` provided to the component.
+       * @returns {ReactNode}
+       */
+      renderValue?: (value: T) => React.ReactNode;
+    }
+  | { displayEmpty: true; multiple?: false; renderValue?: (value: T | '') => React.ReactNode }
```

Comment:
I think we'll have to copy the JSDocs here as well as they don't show up in VSCode otherwise.